src/file_io.jl

```diff
@@ -107,3 +125,26 @@ end
 # FileIO Interface
 fileio_save(f, args...; kwargs...) = save(f.filename, args...; kwargs...)
 fileio_load(f, args...) = load(f.filename, args...)
+
+# Objects were initialially saved inline with the BSON document prior to v4
+# To support larger objects/files we started writing objects to the end of file and only save
+# names and sizes to the objects dict in the bson doc.
+# Since our ideal structure is one dict with object bytes we don't upgrade, but rather
+# extract both formats.
```

Comment:
```suggestion
# Since our ideal structure is one dict with object bytes we don't solve this using the `upgrade`
# function but rather extract both formats here.
```
Can we also add a note in https://github.com/invenia/JLSO.jl/blob/master/src/upgrade.jl
saying that this code exist here, so that we know later when dropping support for v3 that we also need to delete this bit?